examples/drupal_loadtest.rs

```diff
@@ -81,19 +81,27 @@ fn main() {
 }
 
 /// View the front page.
-async fn drupal_loadtest_front_page(client: &mut GooseClient) {
-    let mut response = client.get("/").await;
+async fn drupal_loadtest_front_page(client: &GooseClient) {
+    let mut response = client.get("/", None).await;
 
     // Grab some static assets from the front page.
     match response.response {
         Ok(r) => match r.text().await {
             Ok(t) => {
                 let re = Regex::new(r#"src="(.*?)""#).unwrap();
+                // Collect copy of URLs to run them async
+                let mut urls = Vec::new();
+                let mut futures = Vec::new();
                 for url in re.captures_iter(&t) {
                     if url[1].contains("/misc") || url[1].contains("/themes") {
-                        let _response = client.set_request_name("static asset").get(&url[1]);
+                        urls.push(url[1].to_string());
                     }
                 }
+                for index in 0..urls.len() {
+                    futures.push(client.get(&urls[index], Some("static asset")));
+                }
+                // Asynchronously load all static assets together.
+                futures::future::join_all(futures).await;
```

Comment:
@LionsAd Here's an example of an optimization now possible with an immutable `GooseClient`. We are no longer limited to requesting static assets sequentially.

(In early testing, this means where 10 clients were enough to saturate 100% of our CPU due to blocking, we can now run 10 clients with 85-95% of the CPU) -- this also considerably changes how many logged-in versus logged-out pages are loaded, as we're no longer throttled by blocking.

Reply:
It turns out this "boost" was actually a bug in the test from when we converted it to `async` -- we were never calling `await` for the static assets: but the way we were naming requests this was getting hidden as the wrong requests were getting the `static asset` name. In this PR this is cleaned up, and the name can only apply to the correct request.

This `join_all` actually seems to be a performance regression _except_ in the cases where the client is CPU-starved. We're not optimizing for a CPU-starved load test server, so I removed this change and now we are back to awaiting each static element in sequence.